src/Symfony/Component/Encryption/EncryptionInterface.php

```diff
@@ -0,0 +1,156 @@
+     *
+     * Asymmetric encryption uses a "key pair" ie a public key and a private key.
+     * It is safe to share your public key, but the private key should always be
+     * kept a secret.
+     *
+     * When Alice and Bob wants to communicate securely, they share their public keys with
+     * each other. Alice will encrypt a message with Bob's public key. When Bob
+     * receives the message, he will decrypt it with his private key.
+     *
+     *
+     * <code>
+     *     // Bob:
+     *     $bobKey = $encryption->generateKey();
+     *     $bobPublicOnly = $bobKey->extractPublicKey();
+     *     // Bob sends $bobPublicOnly to Alice
+     *
+     *     // Alice:
+     *     $ciphertext = $encryption->encryptFor('input', $bobPublicOnly);
+     *     // Alice sends $ciphertext to Bob
+     *
+     *     // Bob:
+     *     $message = $encryption->decrypt($ciphertext, $bobKey);
+     * </code>
+     *
+     * @param string       $message      plain text version of the message
+     * @param KeyInterface $recipientKey key with a public key of the recipient
+     *
+     * @return string the output
+     *
+     * @throws EncryptionException
+     * @throws InvalidKeyException
+     */
+    public function encryptFor(string $message, KeyInterface $recipientKey): string;
+
+    /**
+     * Get an encrypted version of the message that only the recipient can read.
+     * The recipient can also verify who sent the message.
+     *
+     * Asymmetric encryption uses a "key pair" i.e. a public key and a private key.
+     * It is safe to share your public key, but the private key should always be
+     * kept secret.
+     *
+     * When Alice and Bob wants to communicate securely, they share their public keys with
+     * each other. Alice will encrypt a message with keypair [ alice_private, bob_public ].
+     * When Bob receives the message, he will decrypt it with keypair [ bob_private, alice_public ].
+     *
+     * <code>
+     *     // Alice:
+     *     $aliceKey = $encryption->generateKey();
+     *     $alicePublicOnly = $aliceKey->extractPublicKey();
+     *     // Alice sends $alicePublicOnly to Bob
+     *
+     *     // Bob:
+     *     $bobKey = $encryption->generateKey();
+     *     $bobPublicOnly = $bobKey->extractPublicKey();
+     *     // Bob sends $bobPublicOnly to Alice
+     *
+     *     // Alice:
+     *     $ciphertext = $encryption->encryptForAndSign('input', $bobPublicOnly, $aliceKey);
+     *     // Alice sends $ciphertext to Bob
+     *
+     *     // Bob:
+     *     $keypairForReceiving = $bobKey->createKeypair($alicePublicOnly)
+     *     $message = $encryption->decrypt($ciphertext, $keypairForReceiving);
+     * </code>
+     *
+     * @param string       $message      plain text version of the message
+     * @param KeyInterface $recipientKey public key of the recipient
+     * @param KeyInterface $senderKey    private key of the sender
+     *
+     * @return string the output
```

Comment:
can't we remove the `@param` and `@return`? (we should wonder the same in all docblocks)
maybe rename the args to `$recipientPublicKey` / `$senderPrivateKey` since that "private"/"public" might be the useful part of the description?

Reply:
> can't we remove the @param and @return?

I want to add some name of the output, previously it said "the JWE output" . 

> maybe rename the args to $recipientPublicKey / $senderPrivateKey

Hm.. For these implementations, you are correct. But I dont have the concept of "public" and "private" key in the current PR. We only have "key" and "public key".